kamailio: init at 5.7.1

mawis opened the pull request:
###### Description of changes

Created new package 'kamailio' which is a SIP server, that is able to handle thousands of call setups per second. It can be used to build large platforms for VoIP and realtime communictions - presence, WebRTC, Instant messaging and other applications.

https://www.kamailio.org/

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Closes #169350

drupol commented on pkgs/servers/sip/kamailio/default.nix:
```diff
@@ -0,0 +1,76 @@
+, stdenv
+}:
+
+stdenv.mkDerivation rec {
```
I think we should use the `finalAttrs` pattern here.

Let me share a couple of useful links regarding this:

1. History: https://discourse.nixos.org/t/avoid-rec-expresions-in-nixpkgs/8293
2. Documentation: https://nixos.org/manual/nixpkgs/unstable/#mkderivation-recursive-attributes
3. Recent example of implementation: https://github.com/NixOS/nixpkgs/compare/17f96f7b978e61576cfe16136eb418f74fab9952..9e6ea843e473d34d4f379b8b0d8ef0425a06defe

mawis replied:
Thanks, I replaced `rec` with the `finalAttrs` pattern.

drupol commented on pkgs/servers/sip/kamailio/default.nix:
```diff
@@ -0,0 +1,76 @@
+
+  src = fetchurl {
+    url = "https://www.kamailio.org/pub/kamailio/5.7.1/src/kamailio-${version}_src.tar.gz";
+    sha256 = "sha256-TkrAj5ohwzJ959UvXNwaMmcIwTRowUitP9/UNQ8D1zw=";
```
```suggestion
    hash = "sha256-TkrAj5ohwzJ959UvXNwaMmcIwTRowUitP9/UNQ8D1zw=";
```

mawis replied:
Thanks, haven't seen that before. I guess I did to much copying from other packages ;)

drupol reviewed (changes requested):
Nice, it's almost perfect.

Your PR must now contains only 2 commits.

1 for adding you as maintainer.
1 for adding the new derivation.

mawis committed:
maintainers: add Matthias Wimmer

mawis committed:
kamailio: init at 5.7.1

Created new package 'kamailio' which is a SIP server, that is able to
handle thousands of call setups per second. It can be used to build
large platforms for VoIP and realtime communictions - presence, WebRTC,
Instant messaging and other applications.

Closes #169350

mawis commented:
> Nice, it's almost perfect.
> 
> Your PR must now contains only 2 commits.
> 
> 1 for adding you as maintainer. 1 for adding the new derivation.

Done

mawis commented:
Thanks for merging.
